internal/service/kyma/deletion/deletion_service.go

```diff
@@ -0,0 +1,71 @@
+package deletion
+
+import (
+	"context"
+	"errors"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/errors/kyma/deletion"
+	"github.com/kyma-project/lifecycle-manager/internal/result"
+)
+
+var ErrUnableToDetermineUsecaseApplicability = errors.New("unable to determine usecase applicability")
+
+type UseCase interface {
+	IsApplicable(ctx context.Context, kyma *v1beta2.Kyma) (bool, error)
+	Execute(ctx context.Context, kyma *v1beta2.Kyma) result.Result
+	Name() result.UseCase
+}
+
+type Service struct {
+	deletionSteps []UseCase
+}
+
+func NewService(
+	setKcpKymaStateDeleting UseCase,
```

Comment:
This is very brittle... There's no compile-time check of these types...
There are 11 arguments here, one can provide the same object 11 times and the compiler will be happy. Any order of these arguments also makes compiler happy.
The only thing that makes this work is the developer being very careful in providing the right objects at the correct positions...
I think we should find an alternative that makes the mentioned mistakes compile-time errors.